html/cross-origin-embedder-policy/header-parsing.https.html

```diff
@@ -0,0 +1,75 @@
+<!DOCTYPE html>
+<html>
+  <meta charset="utf-8">
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+<body>
+<script>
+'use strict';
+function createIframe(t, values) {
+  const parent = document.createElement('iframe');
+  const child = document.createElement('iframe');
+  const params = new URLSearchParams(values.map((value) => ['value', value]));
+  parent.setAttribute('src', `resources/empty-coep.py?${params}`);
+  document.body.appendChild(parent);
+  t.add_cleanup(() => parent.remove());
+
+  return new Promise((resolve, reject) => {
+      parent.onload = resolve;
+      parent.onerror = reject;
+    })
+    .then(() => {
+        child.setAttribute('src', '/common/blank.html');
+        parent.contentDocument.body.appendChild(child);
+        return new Promise((resolve) => child.onload = resolve);
+      })
+    .then(() => child);
+}
+
+[
+  [],
+  [''],
+  ['jibberish'],
+  ['require\u2014corp'], // non-ASCII character (em dash)
+  ['require-corp;'],
+  ['require-corp\0'],
+  ['Require-corp'],
+  ['require-corp; foo=bar'],
+  ['"require-corp"'], // HTTP structured header "string" item
+  ['*cmVxdWlyZS1jb3Jw*'], // HTTP structured header "byte sequence" item
+  ['require-corp require-corp'],
+  ['require-corp,require-corp'],
+  ['require-corp;require-corp'],
+  ['require-corp', 'require-corp'],
+  ['', 'require-corp'],
+  ['require-corp', ''],
+].forEach((values) => {
+  async_test((t) => {
```

Comment:
The conversion between promises and async_test seems pretty confusing and error-prone. (E.g. I think the .then() handler needs to be wrapped in a t.step_func?)

Why not just use promise_test?

Reply:
Looks like I confused this suite of tests with the suite for COOP (where there's no precedence for `promise_test`). I'm happy to use a more modern and predictable API; I'll just mark this as a "long" test to reduce the risk of timeouts from running these in series.

Reply:
There doesn't need to be consistency between the tests. I'd recommend using whichever of `async_test`, `promise_test`, or `test` make the most sense for the given test or series of tests.